extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsStorageDruidModule.java

```diff
@@ -126,5 +130,23 @@ public void configure(Binder binder)
     Binders.taskLogsBinder(binder).addBinding("hdfs").to(HdfsTaskLogs.class);
     JsonConfigProvider.bind(binder, "druid.indexer.logs", HdfsTaskLogsConfig.class);
     binder.bind(HdfsTaskLogs.class).in(LazySingleton.class);
+
+    final String principal = props.getProperty("druid.hdfs.security.kerberos.principal");
+    final String keytab = props.getProperty("druid.hdfs.security.kerberos.keytab");
+    if (!Strings.isNullOrEmpty(principal) && !Strings.isNullOrEmpty(keytab)) {
+      UserGroupInformation.setConfiguration(conf);
+      if (UserGroupInformation.isSecurityEnabled()) {
+        try {
+          if (UserGroupInformation.getCurrentUser().hasKerberosCredentials() == false
+              || UserGroupInformation.getCurrentUser().getUserName() != principal) {
```

Comment:
actually was thinking that it might be the case the job has a different pair of principal/keytab then if the user is logged with a different principal will do the auth again. but to be honest not sure if it is a valid use case. if i bother you i can remove and assume that hdfs and the job should use the same credentials.